pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/ZkBasedTableRebalanceObserver.java

```diff
@@ -284,4 +328,306 @@ public static TableRebalanceProgressStats.RebalanceStateStats getDifferenceBetwe
         (totalSegments == 0) ? 0 : ((double) rebalanceStats._segmentsToRebalance / totalSegments) * 100.0;
     return rebalanceStats;
   }
+
+  /**
+   * Updates the overall progress stats based on the current step's progress stats. This should be called
+   * during the EV-IS convergence trigger to ensure the overall stats reflect the changes as they are made.
+   * @param rebalanceProgressStats the rebalance stats
+   * @param lastStepStats step level stats from the last iteration
+   * @param latestStepStats latest step level stats calculated in this iteration
+   * @return the newly calculated overall progress stats
+   */
+  @VisibleForTesting
+  static TableRebalanceProgressStats.RebalanceProgressStats updateOverallProgressStatsFromStep(
```

Comment:
make this a method of TableRebalanceProgressStats, so we can do `overall.update(last, latest)` iiuc

Reply:
one